docs/core/integration-quality-scale/rules/iqs026.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "IQS026 - Implements diagnostics"
+---
+
+IQS026 is as follows:
+
+Implements diagnostics
+
+## Reasoning
+
+Diagnostics are an easy way to gather data about the integration and can prove useful when debugging an integration.
+
+We consider it a good practice to have the diagnostics implemented.
+
+## Example implementation
+
+In the following example we provide diagnostics which includes data from various sources, such as the configuration and the current state of the integration.
+
+`diagnostics.py`:
+```python
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: MyyConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+
+    return {
+        "device_model": entry.data[CONF_MODEL],
+        "data": entry.runtime_data.data,
+    }
+```
+
+## Additional resources
+
+
+## Exceptions
+
+There are no exceptions to this rule.
```

Comment:
In the past we never had an exclusion, hence I didn't add one now.

But we have the case now where you can have integrations that are practically stateless. For example, I want to add an integration for The movie database, which could be useful for automations in smart home or voice related. But it will only work with services that return a response value. So it will not have any state, and the config entry data will most likely only be the api key.